### DISCUSSION TITLE
Get ABI & contract read/write methods from bytecode

### DISCUSSION THREAD

**Discussion by o-az** [wevm/viem#906] [Question] [answered]
These 3 contracts are written in Yul so Etherscan can't verify & publish the code.
I would like to know what the read and write methods are. Is it possible to get those somehow with viem?

https://docs.axiom.xyz/transparency-and-security/on-chain-zk-verifiers#deployed-verifier-contracts

**Comment by Raiden1411** [comment, marked as answer]
Its probably best that you use a tool like this https://github.com/shazow/whatsabi
